File: aufgabe1/sqareroot/sqrt.py
# calculate the square root of a number using the Newton-Raphson method

def sqrt(x):
    """
    Calculate the square root of a number using the Newton-Raphson method.

    Args:
        x: The number for which the square root is to be calculated.

    Returns:
        The square root of x.
    """
    # Initial guess for the square root
    z = x / 2.0
    # Keep getting a better estimate for the square root
    while abs(x - (z * z)) > 0.0001:
        z -= (z * z - x) / (2 * z)
    return z


# main program
if __name__ == "__main__":
    while True:
        num = input("Enter a number (or 'exit' to quit): ")
        if num.lower() == 'exit':
            break
        else:
            try:
                num = float(num)
                print("Square root of", num, "is", sqrt(num))
            except ValueError:
                print("Invalid input. Please enter a number.")

File: aufgabe1/sqareroot/sqrt.js
// calculate square root of a number using Newton's method

function newton_sqrt(number) {
    // initial guess
    let guess = number / 2;
    let newGuess = (guess + number / guess) / 2;
    while (Math.abs(newGuess - guess) > 0.0001) {
        guess = newGuess;
        newGuess = (guess + number / guess) / 2;
    }
    return newGuess;
}

// run the function reading input from the command line
let args = process.argv.slice(2);
let numbers = args.map(Number);

numbers.forEach(number => {
    console.log(`Square root of ${number} is ${newton_sqrt(number)}`);
});